src/wp-includes/block-template.php

```diff
@@ -235,16 +235,61 @@ function get_the_block_template_html() {
 		return;
 	}
 
+	// Record all post content strings into this list to avoid double parsing.
+	$parsed_post_content = array();
+	add_filter(
+		'render_block_core/post-content',
+		function( $block_content ) use ( &$parsed_post_content ) {
+			$parsed_post_content[] = $block_content;
+			return $block_content;
+		},
+		PHP_INT_MAX
+	);
+
 	$content = $wp_embed->run_shortcode( $_wp_current_template_content );
 	$content = $wp_embed->autoembed( $content );
 	$content = do_blocks( $content );
+
+	// Replace all post content within the template with a temporary placeholder so that the post content is not
+	// unnecessarily processed again. See https://core.trac.wordpress.org/ticket/55996.
```

Comment:
```suggestion
	/*
	 * Replace all post content within the template with a temporary placeholder so that the post content is not
	 * unnecessarily processed again. See https://core.trac.wordpress.org/ticket/55996.
	 */
```
Should use the multiline comment format here.